pkg/util/atomicvalue/atomic_value.go

```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package util
+package atomicvalue
 
 import (
 	"sync"
```

Comment:
While we're refactoring, we should move the HighWaterMark to it's own file in the same package.  and we should rename Check to Update